stdlib/embed/fs.go

```diff
@@ -0,0 +1,119 @@
+package embed
+
+import (
+	"bytes"
+	"compress/gzip"
+	"embed"
+	"io/fs"
+	"io/ioutil"
+	"time"
+)
+
+//go:embed stdlib
+var stdlib embed.FS
+
+var FS fs.FS = &gzipFS{fs: stdlib}
+
+type gzipFS struct {
```

Comment:
It looks like gzipFS implements several of the interfaces in the `io/fs` package, Can you document which interfaces its implementing and why?

Reply:
Yea, I can document this. Honestly, it's a bit more complicated than I thought it would be and it makes me think I did something wrong. I figure it will be easy to fix once I understand the new API better in the future.